src/main/kotlin/com/natpryce/konfig/cli.kt

```diff
@@ -89,10 +89,12 @@ fun parseArgs(args: Array<String>,
             this[opt]?.configKey ?: throw Misconfiguration("unrecognised command-line option $arg")
         
         fun storeNextArg(configNameByOpt: Map<String, CommandLineOption>, opt: String) {
-            i++
-            if (i >= args.size) throw Misconfiguration("no argument for $arg command-line option")
-            
-            properties[configNameByOpt.configNameFor(opt)] = CommandLineProperty(arg, args[i])
+            if (i + 1 >= args.size || args[i+1].startsWith("-")) {
+                properties[configNameByOpt.configNameFor(opt)] = CommandLineProperty(arg, "true")
```

Comment:
I think this would be confusing for non-boolean options.  Is there a way to make this type-safe and work for booleans only?

Reply:
Or, generalise it and make it work for any type?  That way, it could work for enums as well as booleans.

For example: the CommandLineProperty class could have an optional (nullable) default value field.  If it is non-null, that value is used when an explicit argument is not given.

An overload of the CommandLineProperty function for boolean keys could set the default value to true.

Reply:
I thought about providing a PR on my own for this and my solution would have been to introduce a `CommandLineFlag` class that is meant for exactly that. However, by now I think that the best solution would be to interpret every `booleanType` as a flag since nothing else makes sense.